sync_ci/pkg/detect/detect.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/pingcap/ci/sync_ci/pkg/model"
+	"github.com/pingcap/ci/sync_ci/pkg/parser"
+	"reflect"
+	"time"
+)
+
+const searchIssueIntervalStr = "178h"
+const PrInspectLimit = time.Hour * 24 * 7
+
+func GetCasesFromPR(cfg model.Config, startTime time.Time, inspectStartTime time.Time) ([]*model.CaseIssue, error) {
+	cidb, err := SetupCIDB(cfg)
+	baselink := "https://internal.pingcap.net/idc-jenkins/job/%s/%s/display/redirect" // job name / job id
+	if err != nil {
+		return nil, err
+	}
+
+	// Get failed cases from CI data
+	now := time.Now()
+	rows, err := cidb.Raw(model.GetCICaseSql, formatT(inspectStartTime), formatT(now)).Rows()
+	if err != nil {
+		return nil, err
+	}
+	recentRows, err := cidb.Raw(model.GetCICaseSql, formatT(startTime), formatT(now)).Rows()
+	if err != nil {
+		return nil, err
+	}
+
+	caseSet := map[string]map[string][]string{}
+	repoPrCases := map[string]map[string]string{} // repo -> pr -> case
+	//caseBuildLinks := map[string] []string {}
+	for rows.Next() {
+		var rawCase []byte
+		var cases []string
+		var pr string
+		var repo string
+		var jobid string
+		var job string
+		_ = rows.Scan(&repo, &pr, &rawCase, &jobid, &job)
+		_ = json.Unmarshal(rawCase, &cases)
+		for _, c := range cases {
+			if _, ok := repoPrCases[repo]; !ok {
+				repoPrCases[repo] = map[string]string{}
+			}
+			if _, ok := repoPrCases[repo][c]; !ok {
+				repoPrCases[repo][c] = pr
+			} else {
+				if _, ok = caseSet[repo]; !ok {
+					caseSet[repo] = map[string][]string{}
+				}
+				caseSet[repo][c] = append(caseSet[repo][c], fmt.Sprintf(baselink, job, jobid))
+			}
+		}
+	}
+
+	recentCaseSet := map[string]map[string][]string{}
+	for recentRows.Next() {
+		var rawCase []byte
+		var cases []string
+		var pr string
+		var repo string
+		_ = recentRows.Scan(&repo, &pr, &rawCase)
+		_ = json.Unmarshal(rawCase, &cases)
+
+		for _, c := range cases {
+			if _, ok := caseSet[repo]; !ok {
+				continue
+			}
+			if _, ok := caseSet[repo][c]; !ok {
+				continue
+			}
+			if _, ok := recentCaseSet[repo]; !ok {
+				recentCaseSet[repo] = map[string][]string{}
+			}
+			if matched, name := parser.MatchAndParseSQLStmtTest(c); matched {
+				recentCaseSet[repo][name] = caseSet[repo][c]
+			} else {
+				recentCaseSet[repo][c] = caseSet[repo][c]
+			}
+		}
+	}
+
+	// Validate repo cases
+	dbIssueCase, err := SetupCaseIssueDB(cfg)
+	if err != nil {
+		return nil, err
+	}
+	dbGithub, err := SetupGHDB(cfg)
+	if err != nil {
+		return nil, err
+	}
+	// assumed `cases` param has no reps
+	issueCases := []*model.CaseIssue{}
+	for repo, repoCases := range recentCaseSet {
+		for c, v := range repoCases {
+			existedCases, err := dbIssueCase.Raw(model.IfValidIssuesExistSql, c, repo).Rows()
+			if err != nil {
+				return nil, err
+			}
+			if !existedCases.Next() {
+				issueCase := model.CaseIssue{
+					IssueNo:   0,
+					Repo:      repo,
+					IssueLink: sql.NullString{},
+					Case:      sql.NullString{c, true},
+					JobLink:   sql.NullString{v[0], true},
```

Comment:
You should promise the joblink is the latest.

Reply:
Query changed accordingly to put the latest job link on top